annotation-utils/src/main/java/edu/isi/vista/annotationutils/PushAnnotations.kt

```diff
@@ -0,0 +1,145 @@
+import org.eclipse.jgit.lib.ObjectId
+import org.eclipse.jgit.revwalk.RevCommit
+import org.eclipse.jgit.revwalk.RevWalk
+import org.eclipse.jgit.transport.*
+import org.eclipse.jgit.treewalk.CanonicalTreeParser
+import org.eclipse.jgit.util.FS
+import java.text.SimpleDateFormat
+import java.util.*
+
+/**
+ * The goal of this program is to export the annotations, save the restored
+ * documents, compare with the previous commit to see if any changes have been
+ * made, and if so, push those changes to the repo.
+ *
+ * The program takes one argument: a file containing these parameter values:
+ *
+ * <ul>
+ *     <li> {@code exportAnnotationsParams}: the location of the parameter file meant for {@code ExportAnnotations.kt} </li>
+ *     <li> {@code restoreJsonParams}: the location of the parameter file meant for {@code RestoreJson.kt} </li>
+ *     <li> {@code repoPath}: the location of the local repository
+ *     (https://github.com/isi-vista/curated-training-annotation)</li>
+ * </ul>
+ *
+ * This program currently assumes these things:
+ * <ul>
+ *     <li>The annotations will be saved to .../curated-training-annotation/data/exported</li>
+ *     <li>Your ssh private key is saved in ~/.ssh/id_rsa.</li>
+ * </ul>
+ */
+
+
+fun pushNew(repoPath: String) {
+
+    // Source: https://github.com/kittinunf/Kotlin-Playground/blob/master/buildSrc/src/main/java/com/github/kittinunf/gradle/ReleaseToMasterMergeTask.kt
+    class SshConfig(val sshKeyPath: String): TransportConfigCallback {
+        override fun configure(transport: Transport?) {
+            (transport as SshTransport).sshSessionFactory = object: JschConfigSessionFactory() {
+                override fun configure(hc: OpenSshConfig.Host?, session: Session?) {}
+                override fun createDefaultJSch(fs: FS?): JSch = super.createDefaultJSch(fs).apply {
+                    addIdentity(sshKeyPath)
+
+                }
+            }
+        }
+    }
+
+    val git = Git.open(File(repoPath))
+    val config = git.getRepository().getConfig()
+    config.setString(
+            "remote",
+            "origin",
+            "url",
+            "git@github.com:isi-vista/curated-training-annotation.git")
+    config.save()
+
+    val currentDate = Calendar.getInstance(TimeZone.getTimeZone("UTC"))
+    TimeZone.setDefault(TimeZone.getTimeZone("UTC"))
+    val formatter = SimpleDateFormat("yyyy-MM-dd-HH-mm-ss")
+
+    // Currently working on test_branch
+    git.checkout().setName("test_branch").call()
+
+    git.add().addFilepattern("data/").call()
+    val newCommit = git.commit()
+            .setMessage("includes annotations from ${currentDate.getTime()}").call()
+    git.tag().setName(formatter.format(currentDate.getTime())).call()
+
+
+    if (hasDiff(git, newCommit)) {
+        val pushIter = git.push()
+                .setTransportConfigCallback(SshConfig("~/.ssh/id_rsa"))
+                .call()
+        val pushResult = pushIter.iterator().next()
```

Comment:
What would it mean for there to be multiple push results? Is this something we need to be concerned with?

Reply:
Removed `pushIter` and `pushResult`